src/lib/support/BUILD.gn

```diff
@@ -333,6 +333,11 @@ static_library("support") {
   if (chip_config_memory_debug_dmalloc) {
     libs += [ "dmallocthcxx" ]
   }
+
+  if (chip_device_platform == "esp32" || chip_device_platform == "openiotsdk") {
```

Comment:
Including ` chip_device_platform == "openiotsdk"` in this PR is done on purpose?